apis/v1alpha2/httproute_types.go

```diff
@@ -83,9 +83,10 @@ type HTTPRouteSpec struct {
 	// HTTPRoute specified `test.example.com` and `test.example.net`,
 	// `test.example.net` must not be considered for a match.
 	//
-	// If hostnames do not match with the criteria above, then the HTTPRoute is
-	// not accepted, and the implementation must raise an 'Accepted' Condition
-	// with a status of `False` for the target Listener(s).
+	// If hostnames are specified, and none match with the criteria above, then
```

Comment:
```suggestion
	// When hostnames specified in Route and none match with the criteria above, then
```

Reply:
There's a bit of nuance here, it's really only if hostnames are specified on both the Gateway and Route. If they are absent in either place, everything should match. I'll work on clarifying that.

Reply:
Yes, that's what I was trying to capture as well but probably didn't do a good job either.
I was trying to imply gateway's listener hostname with the "criteria above" part of the sentence but that feels indirect as well.

Reply:
Thanks for catching this @hbagdi! I've updated the PR to make this more clear, PTAL.